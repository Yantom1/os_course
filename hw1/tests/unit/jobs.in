sleep 100&
sleep 200&
sleep 300&
sleep 400&
sleep 20&
sleep 600&
sleep 700&
jobs
sleep 800&
sleep 900&
jobs
kill -9 3
jobs
sleep 1000&
jobs
kill -9 10
jobs
sleep 1001&
jobs
fg
CtrlZ
jobs
fg
CtrlZ
jobs
fg
CtrlZ
jobs
fg
CtrlZ
jobs
CtrlC
jobs
sleep 1002
!time.sleep(3)
CtrlZ
jobs
bg
jobs
sleep 3
CtrlC
sleep 3
!time.sleep(5)
# This signals shouldnt do anything
CtrlC
CtrlC
CtrlZ
CtrlZ
jobs
sleep 1100
CtrlZ
sleep 1200
CtrlZ
sleep 1200
CtrlZ
fg 12
CtrlZ
fg 12
CtrlZ
fg 12
CtrlZ
jobs
fg 11
CtrlZ
fg 11
CtrlC
fg 20
fg 21
fg -2
jobs
# Failures
bg 12
bg 20
bg -1
jobs
# should fail
bg 7
bg 8
jobs
kill -9 1
!time.sleep(1)
kill -9 1
jobs
kill -9 2
kill -9 4
# Failures
fg 4
bg 2
fg 1
jobs
sleep 3&
jobs
!time.sleep(5)
jobs
jobs
jobs
!time.sleep(10)
jobs
fg 
CtrlZ
jobs
fg 
CtrlZ
jobs
fg 
CtrlZ
jobs
sleep 777&
jobs
sleep 888
CtrlC
sleep 888
CtrlZ
bg
jobs
# piazza 59
sleep 100&
jobs
fg > example.txt
CtrlZ
echo "I'm the real slim shady"
cat example.txt
quit kill
