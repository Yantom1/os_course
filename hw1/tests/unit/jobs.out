smash> smash> smash> smash> smash> smash> smash> smash> [1] sleep 100& : 44859 1 secs
[2] sleep 200& : 44861 1 secs
[3] sleep 300& : 44863 1 secs
[4] sleep 400& : 44865 0 secs
[5] sleep 20& : 44867 0 secs
[6] sleep 600& : 44869 0 secs
[7] sleep 700& : 44871 0 secs
smash> smash> smash> [1] sleep 100& : 44859 1 secs
[2] sleep 200& : 44861 1 secs
[3] sleep 300& : 44863 1 secs
[4] sleep 400& : 44865 0 secs
[5] sleep 20& : 44867 0 secs
[6] sleep 600& : 44869 0 secs
[7] sleep 700& : 44871 0 secs
[8] sleep 800& : 44874 0 secs
[9] sleep 900& : 44876 0 secs
smash> signal number 9 was sent to pid 44863
smash> [1] sleep 100& : 44859 1 secs
[2] sleep 200& : 44861 1 secs
[4] sleep 400& : 44865 0 secs
[5] sleep 20& : 44867 0 secs
[6] sleep 600& : 44869 0 secs
[7] sleep 700& : 44871 0 secs
[8] sleep 800& : 44874 0 secs
[9] sleep 900& : 44876 0 secs
smash> smash> [1] sleep 100& : 44859 2 secs
[2] sleep 200& : 44861 2 secs
[4] sleep 400& : 44865 1 secs
[5] sleep 20& : 44867 1 secs
[6] sleep 600& : 44869 1 secs
[7] sleep 700& : 44871 1 secs
[8] sleep 800& : 44874 1 secs
[9] sleep 900& : 44876 1 secs
[10] sleep 1000& : 44881 0 secs
smash> signal number 9 was sent to pid 44881
smash> [1] sleep 100& : 44859 2 secs
[2] sleep 200& : 44861 2 secs
[4] sleep 400& : 44865 1 secs
[5] sleep 20& : 44867 1 secs
[6] sleep 600& : 44869 1 secs
[7] sleep 700& : 44871 1 secs
[8] sleep 800& : 44874 1 secs
[9] sleep 900& : 44876 1 secs
smash> smash> [1] sleep 100& : 44859 2 secs
[2] sleep 200& : 44861 2 secs
[4] sleep 400& : 44865 1 secs
[5] sleep 20& : 44867 1 secs
[6] sleep 600& : 44869 1 secs
[7] sleep 700& : 44871 1 secs
[8] sleep 800& : 44874 1 secs
[9] sleep 900& : 44876 1 secs
[10] sleep 1001& : 44886 0 secs
smash> sleep 1001& : 44886
smash: got ctrl-Z
smash: process 44886 was stopped
smash> [1] sleep 100& : 44859 2 secs
[2] sleep 200& : 44861 2 secs
[4] sleep 400& : 44865 1 secs
[5] sleep 20& : 44867 1 secs
[6] sleep 600& : 44869 1 secs
[7] sleep 700& : 44871 1 secs
[8] sleep 800& : 44874 1 secs
[9] sleep 900& : 44876 1 secs
[10] sleep 1001& : 44886 0 secs (stopped)
smash> sleep 1001& : 44886
smash: got ctrl-Z
smash: process 44886 was stopped
smash> [1] sleep 100& : 44859 3 secs
[2] sleep 200& : 44861 3 secs
[4] sleep 400& : 44865 2 secs
[5] sleep 20& : 44867 2 secs
[6] sleep 600& : 44869 2 secs
[7] sleep 700& : 44871 2 secs
[8] sleep 800& : 44874 2 secs
[9] sleep 900& : 44876 2 secs
[10] sleep 1001& : 44886 0 secs (stopped)
smash> sleep 1001& : 44886
smash: got ctrl-Z
smash: process 44886 was stopped
smash> [1] sleep 100& : 44859 3 secs
[2] sleep 200& : 44861 3 secs
[4] sleep 400& : 44865 2 secs
[5] sleep 20& : 44867 2 secs
[6] sleep 600& : 44869 2 secs
[7] sleep 700& : 44871 2 secs
[8] sleep 800& : 44874 2 secs
[9] sleep 900& : 44876 2 secs
[10] sleep 1001& : 44886 0 secs (stopped)
smash> sleep 1001& : 44886
smash: got ctrl-Z
smash: process 44886 was stopped
smash> [1] sleep 100& : 44859 3 secs
[2] sleep 200& : 44861 3 secs
[4] sleep 400& : 44865 2 secs
[5] sleep 20& : 44867 2 secs
[6] sleep 600& : 44869 2 secs
[7] sleep 700& : 44871 2 secs
[8] sleep 800& : 44874 2 secs
[9] sleep 900& : 44876 2 secs
[10] sleep 1001& : 44886 0 secs (stopped)
smash> smash: got ctrl-C
[1] sleep 100& : 44859 3 secs
[2] sleep 200& : 44861 3 secs
[4] sleep 400& : 44865 2 secs
[5] sleep 20& : 44867 2 secs
[6] sleep 600& : 44869 2 secs
[7] sleep 700& : 44871 2 secs
[8] sleep 800& : 44874 2 secs
[9] sleep 900& : 44876 2 secs
[10] sleep 1001& : 44886 0 secs (stopped)
smash> smash: got ctrl-Z
smash: process 44903 was stopped
smash> [1] sleep 100& : 44859 7 secs
[2] sleep 200& : 44861 7 secs
[4] sleep 400& : 44865 6 secs
[5] sleep 20& : 44867 6 secs
[6] sleep 600& : 44869 6 secs
[7] sleep 700& : 44871 6 secs
[8] sleep 800& : 44874 6 secs
[9] sleep 900& : 44876 6 secs
[10] sleep 1001& : 44886 4 secs (stopped)
[11] sleep 1002 : 44903 3 secs (stopped)
smash> sleep 1002 : 44903
smash> [1] sleep 100& : 44859 7 secs
[2] sleep 200& : 44861 7 secs
[4] sleep 400& : 44865 6 secs
[5] sleep 20& : 44867 6 secs
[6] sleep 600& : 44869 6 secs
[7] sleep 700& : 44871 6 secs
[8] sleep 800& : 44874 6 secs
[9] sleep 900& : 44876 6 secs
[10] sleep 1001& : 44886 4 secs (stopped)
[11] sleep 1002 : 44903 3 secs
smash> smash: got ctrl-C
smash: process 44908 was killed
smash> smash> smash: got ctrl-C
smash: got ctrl-C
smash: got ctrl-Z
smash: got ctrl-Z
[1] sleep 100& : 44859 13 secs
[2] sleep 200& : 44861 13 secs
[4] sleep 400& : 44865 12 secs
[5] sleep 20& : 44867 12 secs
[6] sleep 600& : 44869 12 secs
[7] sleep 700& : 44871 12 secs
[8] sleep 800& : 44874 12 secs
[9] sleep 900& : 44876 12 secs
[10] sleep 1001& : 44886 10 secs (stopped)
[11] sleep 1002 : 44903 9 secs
smash> smash: got ctrl-Z
smash: process 44917 was stopped
smash> smash: got ctrl-Z
smash: process 44919 was stopped
smash> smash: got ctrl-Z
smash: process 44921 was stopped
smash> sleep 1100 : 44917
smash: got ctrl-Z
smash: process 44917 was stopped
smash> sleep 1100 : 44917
smash: got ctrl-Z
smash: process 44917 was stopped
smash> sleep 1100 : 44917
smash: got ctrl-Z
smash: process 44917 was stopped
smash> [1] sleep 100& : 44859 14 secs
[2] sleep 200& : 44861 14 secs
[4] sleep 400& : 44865 13 secs
[5] sleep 20& : 44867 13 secs
[6] sleep 600& : 44869 13 secs
[7] sleep 700& : 44871 13 secs
[8] sleep 800& : 44874 13 secs
[9] sleep 900& : 44876 13 secs
[10] sleep 1001& : 44886 11 secs (stopped)
[11] sleep 1002 : 44903 10 secs
[12] sleep 1100 : 44917 0 secs (stopped)
[13] sleep 1200 : 44919 1 secs (stopped)
[14] sleep 1200 : 44921 0 secs (stopped)
smash> sleep 1002 : 44903
smash: got ctrl-Z
smash: process 44903 was stopped
smash> sleep 1002 : 44903
smash: got ctrl-C
smash: process 44903 was killed
smash> smash error: fg: job-id 20 does not exist
smash> smash error: fg: job-id 21 does not exist
smash> smash error: fg: job-id -2 does not exist
smash> [1] sleep 100& : 44859 15 secs
[2] sleep 200& : 44861 15 secs
[4] sleep 400& : 44865 14 secs
[5] sleep 20& : 44867 14 secs
[6] sleep 600& : 44869 14 secs
[7] sleep 700& : 44871 14 secs
[8] sleep 800& : 44874 14 secs
[9] sleep 900& : 44876 14 secs
[10] sleep 1001& : 44886 12 secs (stopped)
[12] sleep 1100 : 44917 1 secs (stopped)
[13] sleep 1200 : 44919 2 secs (stopped)
[14] sleep 1200 : 44921 1 secs (stopped)
smash> sleep 1100 : 44917
smash> smash error: bg: job-id 20 does not exist
smash> smash error: bg: job-id -1 does not exist
smash> [1] sleep 100& : 44859 16 secs
[2] sleep 200& : 44861 16 secs
[4] sleep 400& : 44865 15 secs
[5] sleep 20& : 44867 15 secs
[6] sleep 600& : 44869 15 secs
[7] sleep 700& : 44871 15 secs
[8] sleep 800& : 44874 15 secs
[9] sleep 900& : 44876 15 secs
[10] sleep 1001& : 44886 13 secs (stopped)
[12] sleep 1100 : 44917 2 secs
[13] sleep 1200 : 44919 3 secs (stopped)
[14] sleep 1200 : 44921 2 secs (stopped)
smash> smash error: bg: job-id 7 is already running in the background
smash> smash error: bg: job-id 8 is already running in the background
smash> [1] sleep 100& : 44859 16 secs
[2] sleep 200& : 44861 16 secs
[4] sleep 400& : 44865 15 secs
[5] sleep 20& : 44867 15 secs
[6] sleep 600& : 44869 15 secs
[7] sleep 700& : 44871 15 secs
[8] sleep 800& : 44874 15 secs
[9] sleep 900& : 44876 15 secs
[10] sleep 1001& : 44886 13 secs (stopped)
[12] sleep 1100 : 44917 2 secs
[13] sleep 1200 : 44919 3 secs (stopped)
[14] sleep 1200 : 44921 2 secs (stopped)
smash> signal number 9 was sent to pid 44859
smash> smash error: kill: job-id 1 does not exist
smash> [2] sleep 200& : 44861 17 secs
[4] sleep 400& : 44865 16 secs
[5] sleep 20& : 44867 16 secs
[6] sleep 600& : 44869 16 secs
[7] sleep 700& : 44871 16 secs
[8] sleep 800& : 44874 16 secs
[9] sleep 900& : 44876 16 secs
[10] sleep 1001& : 44886 14 secs (stopped)
[12] sleep 1100 : 44917 3 secs
[13] sleep 1200 : 44919 4 secs (stopped)
[14] sleep 1200 : 44921 3 secs (stopped)
smash> signal number 9 was sent to pid 44861
smash> signal number 9 was sent to pid 44865
smash> smash error: fg: job-id 4 does not exist
smash> smash error: bg: job-id 2 does not exist
smash> smash error: fg: job-id 1 does not exist
smash> [5] sleep 20& : 44867 17 secs
[6] sleep 600& : 44869 17 secs
[7] sleep 700& : 44871 17 secs
[8] sleep 800& : 44874 17 secs
[9] sleep 900& : 44876 17 secs
[10] sleep 1001& : 44886 15 secs (stopped)
[12] sleep 1100 : 44917 4 secs
[13] sleep 1200 : 44919 5 secs (stopped)
[14] sleep 1200 : 44921 4 secs (stopped)
smash> smash> [5] sleep 20& : 44867 17 secs
[6] sleep 600& : 44869 17 secs
[7] sleep 700& : 44871 17 secs
[8] sleep 800& : 44874 17 secs
[9] sleep 900& : 44876 17 secs
[10] sleep 1001& : 44886 15 secs (stopped)
[12] sleep 1100 : 44917 4 secs
[13] sleep 1200 : 44919 5 secs (stopped)
[14] sleep 1200 : 44921 4 secs (stopped)
[15] sleep 3& : 44954 0 secs
smash> [6] sleep 600& : 44869 23 secs
[7] sleep 700& : 44871 23 secs
[8] sleep 800& : 44874 23 secs
[9] sleep 900& : 44876 23 secs
[10] sleep 1001& : 44886 21 secs (stopped)
[12] sleep 1100 : 44917 10 secs
[13] sleep 1200 : 44919 11 secs (stopped)
[14] sleep 1200 : 44921 10 secs (stopped)
smash> [6] sleep 600& : 44869 23 secs
[7] sleep 700& : 44871 23 secs
[8] sleep 800& : 44874 23 secs
[9] sleep 900& : 44876 23 secs
[10] sleep 1001& : 44886 21 secs (stopped)
[12] sleep 1100 : 44917 10 secs
[13] sleep 1200 : 44919 11 secs (stopped)
[14] sleep 1200 : 44921 10 secs (stopped)
smash> [6] sleep 600& : 44869 23 secs
[7] sleep 700& : 44871 23 secs
[8] sleep 800& : 44874 23 secs
[9] sleep 900& : 44876 23 secs
[10] sleep 1001& : 44886 21 secs (stopped)
[12] sleep 1100 : 44917 10 secs
[13] sleep 1200 : 44919 11 secs (stopped)
[14] sleep 1200 : 44921 10 secs (stopped)
smash> [6] sleep 600& : 44869 33 secs
[7] sleep 700& : 44871 33 secs
[8] sleep 800& : 44874 33 secs
[9] sleep 900& : 44876 33 secs
[10] sleep 1001& : 44886 31 secs (stopped)
[12] sleep 1100 : 44917 20 secs
[13] sleep 1200 : 44919 21 secs (stopped)
[14] sleep 1200 : 44921 20 secs (stopped)
smash> sleep 1200 : 44921
smash: got ctrl-Z
smash: process 44921 was stopped
smash> [6] sleep 600& : 44869 33 secs
[7] sleep 700& : 44871 33 secs
[8] sleep 800& : 44874 33 secs
[9] sleep 900& : 44876 33 secs
[10] sleep 1001& : 44886 31 secs (stopped)
[12] sleep 1100 : 44917 20 secs
[13] sleep 1200 : 44919 21 secs (stopped)
[14] sleep 1200 : 44921 0 secs (stopped)
smash> sleep 1200 : 44921
smash: got ctrl-Z
smash: process 44921 was stopped
smash> [6] sleep 600& : 44869 34 secs
[7] sleep 700& : 44871 34 secs
[8] sleep 800& : 44874 34 secs
[9] sleep 900& : 44876 34 secs
[10] sleep 1001& : 44886 32 secs (stopped)
[12] sleep 1100 : 44917 21 secs
[13] sleep 1200 : 44919 22 secs (stopped)
[14] sleep 1200 : 44921 0 secs (stopped)
smash> sleep 1200 : 44921
smash: got ctrl-Z
smash: process 44921 was stopped
smash> [6] sleep 600& : 44869 34 secs
[7] sleep 700& : 44871 34 secs
[8] sleep 800& : 44874 34 secs
[9] sleep 900& : 44876 34 secs
[10] sleep 1001& : 44886 32 secs (stopped)
[12] sleep 1100 : 44917 21 secs
[13] sleep 1200 : 44919 22 secs (stopped)
[14] sleep 1200 : 44921 0 secs (stopped)
smash> smash> [6] sleep 600& : 44869 34 secs
[7] sleep 700& : 44871 34 secs
[8] sleep 800& : 44874 34 secs
[9] sleep 900& : 44876 34 secs
[10] sleep 1001& : 44886 32 secs (stopped)
[12] sleep 1100 : 44917 21 secs
[13] sleep 1200 : 44919 22 secs (stopped)
[14] sleep 1200 : 44921 0 secs (stopped)
[15] sleep 777& : 44970 0 secs
smash> smash: got ctrl-C
smash: process 44973 was killed
smash> smash: got ctrl-Z
smash: process 44975 was stopped
smash> sleep 888 : 44975
smash> [6] sleep 600& : 44869 35 secs
[7] sleep 700& : 44871 35 secs
[8] sleep 800& : 44874 35 secs
[9] sleep 900& : 44876 35 secs
[10] sleep 1001& : 44886 33 secs (stopped)
[12] sleep 1100 : 44917 22 secs
[13] sleep 1200 : 44919 23 secs (stopped)
[14] sleep 1200 : 44921 1 secs (stopped)
[15] sleep 777& : 44970 1 secs
[16] sleep 888 : 44975 1 secs
smash> smash> [6] sleep 600& : 44869 35 secs
[7] sleep 700& : 44871 35 secs
[8] sleep 800& : 44874 35 secs
[9] sleep 900& : 44876 35 secs
[10] sleep 1001& : 44886 33 secs (stopped)
[12] sleep 1100 : 44917 22 secs
[13] sleep 1200 : 44919 23 secs (stopped)
[14] sleep 1200 : 44921 1 secs (stopped)
[15] sleep 777& : 44970 1 secs
[16] sleep 888 : 44975 1 secs
[17] sleep 100& : 44979 0 secs
smash> smash> I'm the real slim shady
smash> sleep 100& : 44979
smash: got ctrl-Z
smash: process 44979 was stopped
smash> smash: sending SIGKILL signal to 11 jobs:
44869: sleep 600&
44871: sleep 700&
44874: sleep 800&
44876: sleep 900&
44886: sleep 1001&
44917: sleep 1100
44919: sleep 1200
44921: sleep 1200
44970: sleep 777&
44975: sleep 888
44979: sleep 100&
